sdk/identity/azure-identity/src/main/java/com/azure/identity/AuthorizationCodeCredential.java

```diff
@@ -82,15 +85,21 @@ public class AuthorizationCodeCredential implements TokenCredential {
         this.cachedToken = new AtomicReference<>();
         this.authCode = authCode;
         this.redirectUri = redirectUri;
+        this.useConfidentialClient = !CoreUtils.isNullOrEmpty(clientSecret);
     }
 
     @Override
     public Mono<AccessToken> getToken(TokenRequestContext request) {
         return Mono.defer(() -> {
             isCachePopulated = isCachePopulated(request);
             if (isCachePopulated) {
-                return identityClient.authenticateWithPublicClientCache(request, cachedToken.get())
-                    .onErrorResume(t -> Mono.empty());
+                if (useConfidentialClient) {
+                    return identityClient.authenticateWithConfidentialClientCache(request, cachedToken.get())
+                        .map(accessToken -> (MsalToken) accessToken);
```

Comment:
Why does this need a cast?

Reply:
The reactor flow requires msal token type to be consistent with downstream. 
The cast helps to satisfy that.